src/runtime-rs/crates/hypervisor/src/device/vsock.rs

```diff
@@ -15,3 +20,70 @@ pub struct VsockConfig {
     /// unix domain socket path
     pub uds_path: String,
 }
+
+#[derive(Debug)]
+pub struct VsockConfig {
+    /// Unique identifier of the device
+    pub id: String,
+
+    /// A 32-bit Context Identifier (CID) used to identify the guest.
+    pub guest_cid: u32,
+
+    /// Vhost vsock fd. Hold to ensure CID is not used by other VM.
+    pub vhost_fd: File,
+}
+
+const VHOST_VSOCK_DEVICE: &str = "/dev/vhost-vsock";
+
+// From <linux/vhost.h>
+// Generate a wrapper function for VHOST_VSOCK_SET_GUEST_CID ioctl.
+// It set guest CID for vsock fd, and return error if CID is already
+// in use.
+const VHOST_VIRTIO: u8 = 0xAF;
```

Comment:
Nit: Although this matches what's defined in the header, I think it would be clearer to name it something like:

```suggestion
const VHOST_VIRTIO_IOCTL: u8 = 0xAF;
```

Reply:
fixed